pkg/controller/backupbucket/controller.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket
+
+import (
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// FinalizerName is the backupbucket controller finalizer.
+	FinalizerName = "extensions.gardener.cloud/backupbucket"
+	// ControllerName is the name of the controller
+	ControllerName = "backupbucket-controller"
+)
+
+// AddArgs are arguments for adding a BackupBucket controller to a manager.
+type AddArgs struct {
+	// Actuator is a BackupBucket actuator.
+	Actuator Actuator
+	// ControllerOptions are the controller options used for creating a controller.
+	// The options.Reconciler is always overridden with a reconciler created from the
+	// given actuator.
+	ControllerOptions controller.Options
+	// Predicates are the predicates to use.
+	// If unset, GenerationChangedPredicate will be used.
+	Predicates []predicate.Predicate
+}
+
+// DefaultPredicates returns the default predicates for a controlplane reconciler.
+func DefaultPredicates(mgr manager.Manager) []predicate.Predicate {
+	return []predicate.Predicate{
+		extensionscontroller.GenerationChangedPredicate(),
+	}
+}
+
+// Add creates a new BackupBucket Controller and adds it to the Manager.
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, args AddArgs) error {
+	args.ControllerOptions.Reconciler = NewReconciler(mgr, args.Actuator)
+	return add(mgr, args.ControllerOptions, args.Predicates)
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, options controller.Options, predicates []predicate.Predicate) error {
+	ctrl, err := controller.New(ControllerName, mgr, options)
+	if err != nil {
+		return err
+	}
+
+	return ctrl.Watch(&source.Kind{Type: &extensionsv1alpha1.BackupBucket{}}, &handler.EnqueueRequestForObject{}, predicates...)
```

Comment:
Please also establish a watch for the referenced secrets (like in the other controllers).

Reply:
hmm. I just added for backupEntry.  Will add it.

Reply:
You haven't added it yet?!

Reply:
I've added already. What else you want to add here?

Reply:
Hm, I refreshed the browser, now I see it. Strange, maybe cache issue in Chrome. Thanks!